File: week6/sentimental-readability/readability.py
import cs50


def main():
    # Prompt user for input string
    while True:
        text = cs50.get_string("Text: ")
        # If input string is not empty, break
        if text != "":
            break

    # Get coleman index and print result
    print(get_coleman_index(text))


# Function to generate & return values for total letters, words and sentences using same loop for efficiency
def get_counts(s):
    # Iterate over chars in string. Increment letter_count every time that char.isalpha() == True (or 1)
    letter_count = sum(1 for char in s if char.isalpha())

    # Get length of word list resulting from splitting the input string
    word_count = len(s.split())

    # Count the number of sentence-ending punctuation marks to get sentence_count
    sentence_count = 0
    for c in s:
        if c == "." or c == "!" or c == "?":
            sentence_count += 1

    # Return all count results
    return letter_count, word_count, sentence_count


# Function to calculate the Coleman Index for a given string
def get_coleman_index(s):
    # Declare and populate return values for all counts using get_counts function
    letter_count, word_count, sentence_count = get_counts(s)

    # Generate values for letters and sentences per 100 words
    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100

    # Evaluate for Coleman Index value
    coleman_index = round(0.0588 * L - 0.296 * S - 15.8)

    # Return the corresponding return value
    if coleman_index < 1:
        return "Before Grade 1"
    elif 1 <= coleman_index <= 16:
        return f"Grade {coleman_index}"
    else:
        return "Grade 16+"


# Run the main function
main()

File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// method signature declatrations
void getCounts(string s);
void getColemanIndex(string s);

// global variables
string text;
float letterCount = 0;
float wordCount = 0;
float sentenceCount = 0;

int main(void)
{
    // prompt user for text
    text = get_string("Text: ");

    // calculate & print coleman index value based on text input
    getColemanIndex(text);
}

// generate values for total letters, words and sentences using same loop
void getCounts(string s)
{
    for (int i = 0; i <= strlen(s); i++)
    {

        // if it's a letter, increment letter count
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letterCount++;
        }

        // if it's a space, increment word count
        if (s[i] == ' ')
        {
            wordCount++;
        }

        // if it's a period or a question / exclamation mark, increment sentence count
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentenceCount++;
        }
    }

    // account for word at the end of a sentence
    wordCount++;
}

// calculate and print coleman index value
void getColemanIndex(string s)
{
    // count and save total letters, words and sentences in text
    getCounts(s);

    // generate values for letters and sentences per 100 words
    float L = (float) letterCount / wordCount * 100;
    float S = (float) sentenceCount / wordCount * 100;

    // evaluate for Coleman Index value
    float colemanIndex = round(0.0588 * L - 0.296 * S - 15.8);

    // print colemanIndex response
    if (colemanIndex < 1)
    {
        printf("%s\n", "Before Grade 1");
    }
    else if (colemanIndex >= 1 && colemanIndex <= 16)
    {
        printf("%s%i\n", "Grade ", (int) colemanIndex);
    }
    else
    {
        printf("%s\n", "Grade 16+");
    }
}